mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -55,86 +56,166 @@ static cl::opt<std::string> outputFilename("o", cl::desc("Output filename"),
                               cl::value_desc("Block size"), cl::init(0));
 
-static cl::opt<int> gridSize("grid_size", cl::desc("Grid size"),
+static cl::opt<int> gridSize("grid_size",
+                             cl::desc("Override grid size for tuning"),
                              cl::value_desc("Grid size"), cl::init(0));
 
-// Set up lowering pipeline.
-// The default lowering pipeline compiles down to GPU dialect.
-// The output of the pipeline can be piped to mlir-rocm-runner for execution.
-//
-// When users specify "-c -target=rocdl", compiles down to LLVM dialect.
-// The output of the pipeline can be piped to mlir-translate for translation to
-// LLVM IR.
-static cl::opt<bool> miopenBuiltinPipeline(
-    "miopen_pipeline", cl::desc("Compile with the specified pipeline"),
-    cl::value_desc("By default, compiles down to GPU dialect. Set "
-                   "-target=rocdl compiles to ROCDL dialect."),
-    cl::init(false));
-
-static cl::alias
-    aliasMIOpenBuiltinPipeline("c", cl::aliasopt(miopenBuiltinPipeline));
-
-static cl::opt<bool>
-    highLevelPipeline("high_level_pipeline",
-                      cl::desc("Compile with the specified pipeline"),
-                      cl::init(false));
-
-static cl::alias aliasHighLevelPipeline("hlp", cl::aliasopt(highLevelPipeline));
-
-static cl::opt<std::string> loweringTargetDialect(
-    "target",
-    cl::desc("By default, compiles down to GPU dialect. Set "
-             "-target=rocdl compiles to ROCDL dialect."),
-    cl::value_desc("Target dialect"), cl::init("gpu"));
-
-static cl::opt<int> deviceNum(
-    "device",
-    cl::desc("Device index on which to run the kernel (only with host code)"),
-    cl::value_desc("Between 0 and number of GPUs on system. "
-                   "Omission leaves current device intact."));
-static cl::alias deviceShort("dev", cl::aliasopt(deviceNum));
-
 namespace test {
 void registerTestDialect(DialectRegistry &);
 } // namespace test
 
-static void populateTuningPipeline(PassManager &pm) {
-  pm.addPass(
-      mlir::miopen::createAffixTuningParametersPass(blockSize, gridSize));
+static LogicalResult
+parsePipeline(StringRef pipeline, llvm::SmallDenseSet<StringRef> &pipelineSet,
+              llvm::SmallDenseSet<StringRef> &pipelineOptions) {
+  size_t start = 0;
+  size_t pos = 0;
+  do {
+    pos = pipeline.find_first_of(",", start);
+    auto opt = pipeline.substr(start, pos).trim();
+    if (opt.empty()) {
+    } else if (opt == "all" || opt == "full") {
+      pipelineSet = pipelineOptions;
+    } else if (pipelineOptions.contains(opt)) {
+      pipelineSet.insert(opt);
+    } else {
+      SmallString<256> opts;
+      for (auto pipeline : pipelineOptions) {
+        if (opts.size())
+          opts += ", ";
+        opts += pipeline;
+      }
+      llvm::errs() << "Invalid pipeline: " << opt << "\n"
+                   << "   Valid options: " << opts << " or full\n";
```

Comment:
This check, together with `kernelPipelineOptions` should not be defined here, but as a validation check before `runMLIRPasses` invoked.

Reply:
Seems like there should be a standard way of doing this but I cannot find a cl::opt way to specify a list of values. This is similar to the opt 'pass-pipeline' but much more simple.